src/step.js

```diff
@@ -87,18 +55,27 @@ export default function step(
     params,
     gradientDifference,
     parameterizedFunction,
+    centralDifference,
   );
   let matrixFunc = matrixFunction(data, evaluatedData);
+  // console.log(' identityt' , identity)
+  // console.log(' gradient 2', identity.add(gradientFunc.mmul(gradientFunc.transpose())));
   let inverseMatrix = inverse(
     identity.add(gradientFunc.mmul(gradientFunc.transpose())),
   );
 
   params = new Matrix([params]);
+  // console.log(
+  //   inverseMatrix
+  //   .mmul(gradientFunc)
+  //   .mmul(matrixFunc)
+  //   .transpose()
+  // );
+
   params = params.sub(
     inverseMatrix
       .mmul(gradientFunc)
       .mmul(matrixFunc)
-      .mul(gradientDifference)
```

Comment:
I don't know how this algorithm exactly works but it is surprising we can remove this line.

Reply:
it is because this gradientDifference value now is inside of gradientFunc due it could have a value for each parameter